### DISCUSSION TITLE
Only Terminal View Like Killercoda Shells or Amazon Lightsail Shell Pages

### DISCUSSION THREAD

**Discussion by alperreha** [eclipse-theia/theia#12831] [Q&A] [answered]
Hi all,
I am grateful for the such a great tool. Thanks for your awesome works :) .  
  
I am just getting started to using Theia ide on my project. I need to use Theia-terminal as a iframe on my website page. But in [minimal setup](https://github.com/vince-fugnitto/theia-minimal/tree/main) and main monorepo example repositories setup instruction includes editor-fileexplorer-git-someextensions with "theia build" command. But i want to show only terminal view on my page.For this reason I want to open and attach only terminal view width 100% and height 100% page.  
  
Someone asked that before [Here](https://github.com/eclipse-theia/theia/issues/5663) But i need opposite of that question. How can i build Theia ide project that only displays machine terminal on port like 3030 (no editor or any other plugins needed.). 
Thanks.  

I also tried solution on previous issues like [open terminal by default](https://github.com/eclipse-theia/theia/discussions/10934) but it doesn't work  😞.  
I mean i want to create a view like Killercoda playground right section like shell view or Amazon Lightsail EC2 instance connect page SSH Web Terminal fullscreen like page.  
  
I used package.json likee below:  
```json  
{
  "name": "theia-minimal",
  "keywords": [
    "theia-extension"
  ],
  "version": "0.0.0",
  "files": [
    "lib",
    "src"
  ],
  "dependencies": {
    "@theia/core": "latest",
    "@theia/navigator": "latest",
    "@theia/outline-view": "latest"
  },
  "devDependencies": {
    "rimraf": "latest",
    "typescript": "latest"
  },
  "scripts": {
    "prepare": "yarn run clean && yarn run build",
    "clean": "rimraf lib",
    "build": "tsc",
    "watch": "tsc -w"
  },
  "theiaExtensions": [
    {
      "frontend": "lib/browser/theia-minimal-frontend-module"
    }
  ]
}

```  

  
And I tried to write a plugin for auto open terminal by default but i didn't work :(.  
```ts
// theia-minimal/src/browser/theia-minimal-frontend-module.ts
import { FrontendApplicationContribution } from "@theia/core/lib/browser/frontend-application";
import { TerminalService } from "@theia/terminal/lib/browser/base/terminal-service";
import { ContainerModule, inject } from "@theia/core/shared/inversify";

export class CustomFrontendApplicationContribution
  implements FrontendApplicationContribution
{
  @inject(TerminalService) private readonly terminalService: TerminalService;
  async initializeLayout(): Promise<void> {
    // Open a terminal in the bottom panel
    const terminal = await this.terminalService.newTerminal({});
    await terminal.start();
    this.terminalService.open(terminal, {
      widgetOptions: { area: "right" },
    });
  }
}

export default new ContainerModule((bind, _unbind, _isBound, rebind) => {
  // Bind it like this
  bind(FrontendApplicationContribution).toService(
    CustomFrontendApplicationContribution
  );
});
```

**Comment by msujew** [comment, marked as answer]
Hey @AlperRehaYAZGAN,

I don't think that Theia really fits your usecase here. Using Theia just for the terminal sounds very much overengineered to me. Have you taken a look at [xterm.js](http://xtermjs.org/) yet? That's the library that's Theia using for terminal rendering. Xterm + a bit of glue code is probably a better solution than trying to wrangle Theia into a terminal-only shape.

There's by the way an error in your binding code:

```ts
export default new ContainerModule((bind, _unbind, _isBound, rebind) => {
  bind(CustomFrontendApplicationContribution).toSelf().inSingletonScope(); // <-- This needs to happen first
  bind(FrontendApplicationContribution).toService(CustomFrontendApplicationContribution);
});
```

**Comment by MrZmcH** [comment]
@msujew If I close the terminal manually, can I reopen it through terminalService?

**Reply by msujew** [reply]
The terminal process ends when the terminal is closed. You cannot "reopen" a closed terminal. You can only open a new terminal.

Next time, please create a new GitHub discussion for such questions since your question is basically completely unrelated to the discussion.
